shared/src/main/java/com/couchbase/lite/Expression.java

```diff
@@ -342,48 +426,54 @@ public String toString() {
         return String.format(Locale.ENGLISH, "%s[json=%s]", getClass().getSimpleName(), asJSON());
     }
 
-    protected Object jsonValue(Object value) {
-        if (value instanceof Expression)
-            return ((Expression) value).asJSON();
-        else
-            return value;
-    }
-
     abstract Object asJSON();
 
+    static final class ValueExpression extends Expression {
+        private Object value;
+
+        ValueExpression(Object value) {
+            this.value = value;
```

Comment:
We need to validate the type of the Object here. Currently we support String, Number, int, long, float, double, boolean, Date, and null.